emuu-client/src/components/upload/UploadButton.js

```diff
@@ -19,82 +27,167 @@ const theme = createTheme({
     },
   },
 });
+function CircularProgressWithLabel(props) {
+  return (
+    <Box sx={{ position: "relative", display: "inline-flex" }}>
+      <CircularProgress variant="determinate" {...props} />
+      <Box
+        sx={{
+          top: 0,
+          left: 0,
+          bottom: 0,
+          right: 0,
+          minWidth: 150,
+          position: "absolute",
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+        }}
+      >
+        <Typography variant="subtitle" component="div" color="black">
+          {`${Math.round(props.value)}%`}
+        </Typography>
+      </Box>
+    </Box>
+  );
+}
 
+CircularProgressWithLabel.propTypes = {
+  value: PropTypes.number.isRequired,
+};
 
 function FileUpload() {
- //use state for registration variables
-  const [videoTitle, setVideoTitle] = useState("")
-  const [videoDescription, setVideoDescription] = useState("")
-  const[videoTag, setVideoTag] = useState("")
-  const[videoDate,setVideoDate] = useState("")
+  //use state for registration variables
+  const [videoTitle, setVideoTitle] = useState("");
+  const [videoDescription, setVideoDescription] = useState("");
+  const [videoTag, setVideoTag] = useState("");
+  const [videoDate, setVideoDate] = useState("");
+  const [videoUrl, setVideoUrl] = useState("");
+  const [userName, setUserName] = useState("");
+
+  //upload data structure
+  const uploadData = {
+    user_userName: userName,
+    video_title: videoTitle,
+    video_description: videoDescription,
+    video_gameTags: videoTag,
+    video_url: videoUrl,
+  };
+
+  //Gets user authentication
+  const auth = getAuth();
+  const user = auth.currentUser;
+
+  //Gets the RAWG api data for game database
+  getData();
```

Comment:
I'll remove this function since I worked on it. Game API isn't working as intended, should remove all correspondence of it.